src/Core/Models/Business/UserLicense.cs

```diff
@@ -26,7 +26,9 @@ public UserLicense()
             Premium = user.Premium;
             MaxStorageGb = user.MaxStorageGb;
             Issued = DateTime.UtcNow;
-            Expires = subscriptionInfo?.UpcomingInvoice?.Date?.AddDays(7);
+            Expires = subscriptionInfo?.UpcomingInvoice?.Date != null ?
+                subscriptionInfo.UpcomingInvoice.Date?.AddDays(7) :
```

Comment:
`Date?.AddDays(7)`, the `null` conditional is not necessary on this line since you've already checked that it's not `null`. You can remove the `?` from this expression.